frontend/public/components/secrets/create-secret.tsx

```diff
@@ -72,17 +91,25 @@ class BaseEditSecret_ extends SafetyFirst<BaseEditSecretProps_, BaseEditSecretSt
     );
   }
   render () {
-    const title = `${this.props.titleVerb} ${_.upperFirst(this.state.secretType)} Secret`;
+    const title = `${this.props.titleVerb} ${_.upperFirst(this.state.secretTypeAbstraction)} Secret`;
     const { saveButtonText } = this.props;
 
-    const explanation = 'Webhook secrets allow you to authenticate a webhook trigger.';
-    const subform = <WebHookSecretSubform onChange={this.onDataChanged.bind(this)} stringData={this.state.stringData} />;
+    let explanation = null;
```

Comment:
If you can `const` and separate the logic on these, I would:

```JavaScript 
const isWebhook = this.state.secretTypeAbstraction === 'webhook'; 
const explanation = isWebhook 
  ? 'Webhook secrets allow you to authenticate a webhook trigger.'
  : 'Source secrets allow you to authenticate against the SCM server.';
const subform = isWebhook
  ? <WebHookSecretSubform onChange={this.onDataChanged.bind(this)} stringData={this.state.stringData} />
  : <SourceSecretSubform onChange={this.onDataChanged.bind(this)} stringData={this.state.stringData} />;
```

Reply:
There are going to be a bunch more subforms, so this won't scale. We'll eventually have a switch statement for several types.

Reply:
Thats fine.  A `switch` or an `object` lookup works for me.